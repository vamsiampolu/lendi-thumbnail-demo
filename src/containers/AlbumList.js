import React from 'react'
import { Menu } from 'semantic-ui-react'
import { Link } from 'react-router'
import { connect } from 'react-redux'
import uniq from 'lodash.uniq'

export function AlbumItem(props) {
  return (<Menu.Item><Link to={"/albums/" + props.albumId}>Album {props.albumId}</Link></Menu.Item>)
}

export function SimpleAlbumMenu(props) {
  if(props.albums && props.albums.length) {
    const albumItems = props.albums.map(albumId  => (<AlbumItem key={albumId} albumId={albumId} />))
    return (<Menu pointing vertical>{albumItems}</Menu>)
  } else {
    return null;
  }

}

const mapStateToProps = state => {
  if(state.isFetching) {
    return {albums: [] }
  }
  const albumIds = uniq(state.albums.map(item => item.albumId))
  return {albums: albumIds }
}
const AlbumMenu = connect(mapStateToProps)(SimpleAlbumMenu)
export default AlbumMenu
