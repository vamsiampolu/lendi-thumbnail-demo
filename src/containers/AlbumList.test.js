import React from 'react'
import renderer from 'react-test-renderer'
import { SimpleAlbumMenu } from './AlbumList'

describe('SimpleAlbumMenu', () => {
  it('renders correctly when albumItems do not exist', () => {
    const props = {
      albums: []
    }
    const component = renderer.create(<SimpleAlbumMenu {...props} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when albumItems exist', () => {
    const props = {
      albums: ['1','2','3']
    }

    const component = renderer.create(<SimpleAlbumMenu {...props} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
