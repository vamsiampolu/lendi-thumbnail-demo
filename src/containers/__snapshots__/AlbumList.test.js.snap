exports[`SimpleAlbumMenu renders correctly when albumItems do not exist 1`] = `null`;

exports[`SimpleAlbumMenu renders correctly when albumItems exist 1`] = `
<div
  className="ui pointing vertical menu">
  <div
    className="item"
    onClick={[Function]}>
    <a
      onClick={[Function]}
      style={Object {}}>
      Album 
      1
    </a>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <a
      onClick={[Function]}
      style={Object {}}>
      Album 
      2
    </a>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <a
      onClick={[Function]}
      style={Object {}}>
      Album 
      3
    </a>
  </div>
</div>
`;
